pkg/webhooks/ingresscontroller/ingresscontroller.go

```diff
@@ -149,12 +176,151 @@ func (wh *IngressControllerWebhook) authorized(request admissionctl.Request) adm
 		}
 	}
 
-	ret = admissionctl.Allowed("IngressController operation is allowed")
+	/* TODO:
+	 * - HypershiftEnabled is currently set to false/disabled.
+	 *   If HCP is to be enabled for allowed source ranges, should this part of a 2nd ingress validator to
+	 *   allow separation of validations between cluster install types? ...or if there's a reliable run time method available
+	 *   to this validator to determine classic vs hcp they can remain in the single webhook(?)
+	 * - Classic vs HCP could likely share some of the network funcions, but will need slightly
+	 *   different logic as well as permissions fetching the network config info from different
+	 *   source  (configmap) locations and config formats(?).
+	 */
+	// Only check for machine cidr in allowed ranges if creating or updating resource...
+	reqOp := request.AdmissionRequest.Operation
+	if reqOp == admissionv1.Create || reqOp == admissionv1.Update {
+		//TODO: Will these need to iterate over more than just the default IngressController config?
+		if ic.ObjectMeta.Name == "default" && ic.ObjectMeta.Namespace == "openshift-ingress-operator" {
+			ret := wh.checkAllowsMachineCIDR(ic.Spec.EndpointPublishingStrategy.LoadBalancer.AllowedSourceRanges)
+			ret.UID = request.AdmissionRequest.UID
+			if !ret.Allowed {
+				log.Info("Error checking minimum AllowedSourceRange", "err", ret.AdmissionResponse.String())
+			}
+			return ret
+		}
+	}
+	log.Info("############# DEBUG LOG: IngressController operation is allowed ###########")
+	ret = admissionctl.Allowed("IngressController operation is allowed, machineCIDR n/a")
 	ret.UID = request.AdmissionRequest.UID
 
 	return ret
 }
 
+func (wh *IngressControllerWebhook) getMachineCIDR() (net.IP, *net.IPNet, error) {
+	if wh.machineCIDRIP == nil || wh.machineCIDRNet == nil {
+		instConf, err := wh.getClusterConfig()
+		if err != nil {
+			log.Error(err, "Failed to fetch machineCIDR", "namespace", installConfigNamespace, "configmap", installConfigMap)
+			return nil, nil, err
+		}
+		if instConf == nil {
+			err := fmt.Errorf("can not fetch machineCIDR from empty '%s' install config", installConfigMap)
+			log.Error(err, "getMachineCIDR failed to find CIDR value")
+			return nil, nil, err
+		}
+		if len(instConf.Networking.MachineCIDR) <= 0 {
+			err := fmt.Errorf("empty machineCIDR string value parsed from '%s' install config", installConfigMap)
+			log.Error(err, "getMachineCIDR found empty CIDR value")
+			return nil, nil, err
+		}
+		machIP, machNet, err := net.ParseCIDR(string(instConf.Networking.MachineCIDR))
```

Comment:
```suggestion
		_, machNet, err := net.ParseCIDR(string(instConf.Networking.MachineCIDR))
```

Do we actually need both the IP and network here? Seems like we could get by on just the network, no?

Reply:
Thanks Trevor, I think you're correct. For the validations we really only need the machNet (ie: net.ipNet) .network() value and mask size. This is mainly for debug and can be removed.
I had originally left the machIP (ie net.IP) mostly for debug purposes so it would might be easier to map a logged message or response error to the actual value found in the config map.  This might not be a practical scenario, and/or reason to keep this around though? 
ie if it's possible to build a cluster with a full IP ie: '192.168.1.1/8' (maybe accidentally?).  We'll handle and log this as '192.0.0.0/8'.  Which might make debug a little harder when searching for the stored value?

Reply:
Updated to only using ipNet in latest commit.